packages/eslint-plugin/tests/rules/no-duplicate-type-constituents.test.ts

```diff
@@ -0,0 +1,657 @@
+import type { TSESLint } from '@typescript-eslint/utils';
+
+import type {
+  MessageIds,
+  Options,
+} from '../../src/rules/no-duplicate-type-constituents';
+import rule from '../../src/rules/no-duplicate-type-constituents';
+import { getFixturesRootDir, RuleTester } from '../RuleTester';
+
+const rootPath = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: rootPath,
+    project: './tsconfig.json',
+  },
+});
+
+const valid = (operator: '|' | '&'): TSESLint.ValidTestCase<Options>[] => [
```

Comment:
[Cleanup] You know, I've really started to sour on generating tests. It's harder to read and harder to perform static analysis on the strings. Let's just write out the tests explicitly.

One nice thing about this rule is that the logic is the same for intersections and unions. So I don't think you need to duplicate test cases for each of them. Just write a few to make sure that both are accounted for, then any edge case tests for constituent checking logic can be just written once. (and wanting to generate tests for combinations of edge cases is a smell to me in that way!)

Reply:
Understood!
I have rewritten the test in naive form (test generation is not used.). Basically, I am testing with `Union`, but I have included some `Intersection` stuff.